pkgs/by-name/fu/furmark/package.nix

```diff
@@ -0,0 +1,72 @@
+{ autoPatchelfHook
+, copyDesktopItems
+, fetchzip
+, lib
+, libGL
+, libGLU
+, makeDesktopItem
+, makeWrapper
+, stdenv
+, vulkan-loader
+}:
+
+let
+  version = "2.0.16";
+  sources = {
+    "x86_64-linux" = {
+      url = "https://gpumagick.com/downloads/files/2024/furmark2/FurMark_${version}_linux64.zip";
+      hash = "sha256-c/SChpIK2dpVEHI2n1rxk0Xq6hGFV1EKGsvZ8Ke+wyo=";
+    };
+    "i686-linux" = {
+      url = "https://gpumagick.com/downloads/files/2024/furmark2/FurMark_${version}_linux32.zip";
+      hash = "sha256-yXd90FgL3WbTga5x0mXT40BonA2NQtqLzRVzn4s4lLc=";
+    };
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "furmark";
+  inherit version;
+
+  src = fetchzip (sources.${stdenv.system});
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    libGL
+    libGLU
+    makeWrapper
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/opt/furmark
```

Comment:
unfortunately yes, it wants some files from the .zip in the same directory as executable.

unless if I should name this directory in a different way? I'm just used to `opt` from other distros.

Reply:
Usually there is few to no reason to use `/opt` in Nixpkgs, since each application goes to its own dedicated root directory.

I am asking in the sense of "the upstream app searches for a directory inside /opt".

Reply:
How would you do it then? Would just $out/furmark be fine? I don't think just $out/ would be a good target as we have bin + share there.

Reply:
> I don't think just $out/ would be a good target as we have bin + share there.

Why would that be not good?

Reply:
idk, so - would you prefer if everything got unpacked into $out and furmark + FurMark_GUI would be in $out/bin?

Reply:
The binaries should definitely be in $out/bin, the libs in $out/lib and the rest probably in $out/share/furmark

Reply:
will do, thanks!

Reply:
> $out/bin, the libs in $out/lib
not doable, again, tried experimenting with wrapping, patchelfing - binaries really want to be in a directory with everything else :(